File: 20111/lp/pstree/tree.php
<?php

/**
 * Árvore Binária
 * 
 * @author Wanderson Henrique Camargo Rosa
 */
class Tree
{
    /**
     * Elemento Raiz
     * @var Node
     */
    protected $_root;

    /**
     * Configura o Elemento Raiz
     * @param Node $root Elemento para Configuração
     * @return Tree Próprio Objeto para Encadeamento
     */
    public function setRoot(Node $root)
    {
        $this->_root = $root;
        return $this;
    }

    /**
     * Informa o Elemento Raiz
     * @return Node Elemento Solicitado
     */
    public function getRoot()
    {
        return $this->_root;
    }

    /**
     * Insere um Valor na Árvore
     * @param mixed $content Conteúdo para Inserção
     * @return Tree Próprio Objeto para Encadeamento
     */
    public function insert($content)
    {
        $node = new Node($content);
        $root = $this->getRoot();
        if ($root === NULL) {
            $this->setRoot($node);
        } else {
            $root->insert($node);
        }
        return $this;
    }
}

File: trunk/20121/compilers/ast/tree.cpp
#include <string>
#include "tree.h"
namespace AST {

    // Construtor
    Tree::Tree(Node* root) {
        // Configuração
        this->_setRoot(root);
    };

    // Configuração de Raiz
    Tree* Tree::_setRoot(Node* root) {
        // Configuração
        this->_root = root;
        // Encadeamento
        return this;
    };

    // Apresentação de Raiz
    Node* Tree::getRoot() {
        // Apresentação
        return this->_root;
    };

    // Conversão
    std::string Tree::toString() {
        // Apresentação
        return this->getRoot()->toString(0);
    };

};

